### PR TITLE
Replace <CASE_NAME> in path with batch id

### PR DESCRIPTION
**Pull request by pgdr**
Make the BatchSimulator replace occurrences of `<CASE_NAME>` in paths with actual case names.

We slugify the paths in case they should contain slashes.

### CODE REVIEW

**Comment by joakim-hove** [comment]
The code you are touching now - i.e. the `<xxx>` replacements - is one of the most notoriously crappy parts of libres/ert - a test would be good!

File: python/python/res/simulator/batch_simulator.py [hunk 1]
```diff
@@ -3,6 +3,13 @@
 from res.enkf import ResConfig, ErtRunContext, EnKFMain, EnkfConfigNode, EnkfNode, NodeId
 from .batch_simulator_context import BatchContext
 
+def _slug(entity):
+    entity = ' '.join(str(entity).strip().split())
+    for i in range(len(entity)):
```
Comment (markusdregi) [inline python/python/res/simulator/batch_simulator.py]: Couldn't the for-loop be replaced by something along the lines of:
`return map(lambda x : x if x.isalnum() else '_', entity)`?
Possibly a bit more Pythonic?

File: python/python/res/simulator/batch_simulator.py [hunk 2]
```diff
@@ -95,6 +102,7 @@ def start(self, case_name, controls):
         batch complete before you start a new batch.
         """
         ens_config = self.res_config.ensemble_config
+        self.ert.addDataKW("<CASE_NAME>", _slug(case_name))
```
Comment (markusdregi) [inline python/python/res/simulator/batch_simulator.py:102]: I would like that we make `<CASE_NAME>` a part of the API of the `BatchSimulator` by documenting the behavior.

**Comment by pgdr** [comment]
> The code you are touching now - i.e. the `<xxx>` replacements - is one of the most notoriously crappy parts of libres/ert - a test would be good!

I agree that tests are always good, but running `start` is not within the scope of a unit test, nor should we test the actual string replace functionality here.

I can do as Markus asks, documenting the behaviour, but other than that, testing the `start` functionality is not suitable for a unit test.  Thus, we have to fall back to a sort of integration test.  If we proceed along those lines, how would you like that to be set up?

**Comment by joakim-hove** [comment]
> nor should we test the actual string replace functionality here.

Well - not exactly the actual low-level string replacement; that should certainly not be tested here. What has created problems numerous times in the past is related to ordering of the type *this* function must be called after *that* function which .... and the `BatchSimulator` is a new way of running, i.e. unfortunately again exposed to the possibility of these f.ups.

> If we proceed along those lines, how would you like that to be set up?

Let me look around a bit; if it is reasonably simple we can add a test - otherwise I am happy to merge as is.

**Comment by joakim-hove** [comment]
> Let me look around a bit; if it is reasonably simple we can add a test - otherwise I am happy to merge as is.

OK - my suggestion is to modify [this test](https://github.com/Statoil/libres/blob/master/python/tests/res/simulator/test_batch_sim.py). If the config file which is referenced has a `RUNPATH` with `<CASE_NAME>` in you can just verify that runpath directory - with an expanded `<CASE_NAME>` actually exists when the simulation has stopped [here](https://github.com/Statoil/libres/blob/master/python/tests/res/simulator/test_batch_sim.py)

**Review by joakim-hove — COMMENTED** [review]
All good!

**Review by joakim-hove — APPROVED** [review]

**Pull request merged**